pgmpy/factors/Factor.py

```diff
@@ -76,6 +76,9 @@ def __init__(self, variables, cardinality, values):
         >>> phi
         <Factor representing phi(x1:2, x2:2, x3:2) at 0x7f8188fcaa90>
         """
+        if not all(isinstance(value, int) or isinstance(value, float) for value in values):
+            raise TypeError("Values: can contain type int or type float elements only")
+
```

Comment:
@yashu-seth Since we are converting the values in a numpy array in the next line. You can move this check into the next line and check for the dtype of the array. That would be much faster than this approach.